File: Chloe Gwon/2022-06-07-Best-Time-to-Buy-and-Sell-Stock.java
class Solution {
    public int maxProfit(int[] prices) {

        int maxProfit = 0;
        int l = 0;
        int r = 1;
        
        while (r < prices.length){
            if (prices[l] < prices[r]){
                int profit = prices[r] - prices[l];
                if (profit > maxProfit){
                    maxProfit = profit;
                }
                r++;
            } else{
                l = r;
                r += 1;
            }
        }
        return maxProfit;
    }
}

File: Jonghyeok Kim/2022-06-07-best-time-to-buy-and-sell-stock.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_num, max_profit = prices[0], 0
        for idx, num in enumerate(prices):
            min_num = min(min_num, num)
            max_profit = max(num-min_num, max_profit)
        return max_profit
